tests/test_external_tests.py

```diff
@@ -1,90 +1,121 @@
-    test = tests.Md5ChecksumTest(target)
-    test_status = test.get_status()
-    assert test_status == TestStatus.FAIL
+        (tmp_path / exit_code.name).symlink_to(exit_code)
+        ExternalTestMixin._find_process_outputs(tmp_path)
 
 
-def test_that_the_json_load_test_works_on_a_correct_file(test_targets):
-    target = test_targets["jsonld"]
-    test = tests.JsonLoadTest(target)
-    test_status = test.get_status()
-    assert test_status == TestStatus.PASS
+def test_that_a_process_can_be_serialized_and_deserialized():
+    process = Process("foo:bar", ["python"])
+    process_dict = process.to_dict()
+    process_from_dict = Process.from_dict(process_dict)
+    assert process_dict == process_from_dict.to_dict()
 
 
-def test_that_the_json_load_test_works_on_incorrect_files(test_targets):
-    target = test_targets["good"]
-    test = tests.JsonLoadTest(target)
-    test_status = test.get_status()
-    assert test_status == TestStatus.FAIL
+# TODO Make changes to fully support Docker-enabled tests in macOS once it is possile
+def docker_enabled_test(func):
+    """Marks Docker-enabled tests to only run in Linux environments."""
+    return pytest.mark.skipif(
+        "linux" not in sys.platform.lower(),
+        reason="Docker-enabled tests only run in Linux",
+    )(func)
+
+
+class DockerExecutor:
+    """Class for executing a command in a docker container."""
+
+    def __init__(self, image: str, command: str, file_path: str):
+        """Initialize the class."""
+        self.image = image
+        self.command = self.format_command_for_sh(command)
+        self.local_path = file_path
+        self.container_path = os.path.join("/", file_path.split("/")[-1])
+
+    def format_command_for_sh(self, command):
+        """Format the command for sh -c.
+        Standardizes command execution across containers."""
+        escaped_command = shlex.quote(command)
+        formatted_command = f"sh -c {escaped_command}"
+        return formatted_command
+
+    def execute(self):
+        """Execute the command in a docker container."""
+        client = docker.from_env()
+        container = client.containers.run(
+            image=self.image,
+            command=self.command,
+            detach=True,
+            stdout=True,
+            stderr=True,
+            volumes={self.local_path: {"bind": self.container_path, "mode": "ro"}},
+            working_dir="/",
+        )
+        self.exit_code = str(container.wait()["StatusCode"])
+        self.std_out = container.logs(stdout=True, stderr=False).decode("utf-8")
+        self.std_err = container.logs(stdout=False, stderr=True).decode("utf-8")
 
 
-def test_that_the_jsonld_load_test_works_on_a_correct_file(test_targets):
-    target = test_targets["jsonld"]
-    test = tests.JsonLdLoadTest(target)
-    test_status = test.get_status()
-    assert test_status == TestStatus.PASS
+def test_that_the_libtiff_info_test_command_is_produced(test_targets):
+    target = test_targets["tiff"]
```

Comment:
nit: I could see these static strings through the test being an enum of the possible valid values you could have: https://docs.python.org/3/library/enum.html

Reply:
I agree. I already created a [ticket](https://sagebionetworks.jira.com/browse/ORCA-288) for reworking the test file names because I find them to be confusing. I can explore this option as part of that task.